### PR TITLE
Added e2e tests for dependency isolation

### PR DESCRIPTION
**Pull request by gavin-aguiar**
## Description
Added the following e2e tests for dependency isolation

- When customer has an older az func version
- When customer has a new az func version
- Minor changes to grpc and protobuf e2e tests

Fixes #

---

### PR information

- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which has an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] If applicable, the PR references the bug/issue that it fixes in the description.
- [x] New Unit tests were added for the changes made and CI is passing.

### Quality of Code and Contribution Guidelines
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-functions-python-worker/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: tests/endtoend/test_dependency_isolation_functions.py [hunk 1]
```diff
@@ -108,14 +111,96 @@ def test_paths_resolution(self):
...
             libraries['proto.expected.version'], libraries['proto.version']
         )
+
         self.assertEqual(
             libraries['grpc.expected.version'], libraries['grpc.version']
         )
+
+
+class TestOlderVersionOfAzFuncDependencyIsolationOnDedicated(
+        testutils.WebHostTestCase):
```
Comment (vrdmr) [inline tests/endtoend/test_dependency_isolation_functions.py:130]: nit - Can you add pydoc comments here describing what the test does.

**Review by vrdmr — APPROVED** [review]
nit changes requested. Once you make the changes, please go ahead and merge the PR.

LGTM

File: tests/endtoend/test_dependency_isolation_functions.py (hunk 1 above)
Comment (vrdmr) [reply tests/endtoend/test_dependency_isolation_functions.py:130]: And same for other tests as well - as these tests are a little involved and would need some context.

E.g.: https://github.com/Azure/azure-functions-python-worker/blob/dev/tests/endtoend/test_eventhub_functions.py#L11-L18

Comment (gavin-aguiar) [reply tests/endtoend/test_dependency_isolation_functions.py:132]: Added pydocs

**Pull request merged**